packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -141,19 +172,84 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
             assumedBy: new cdk.ServicePrincipal('codedeploy.amazonaws.com'),
             managedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole'],
         });
+        this._autoScalingGroups = (props && props.autoScalingGroups) || [];
+        this.addCodeDeployAgentInstallUserData(props && props.agentInstallationType);
 
         const resource = new cloudformation.DeploymentGroupResource(this, 'Resource', {
             applicationName: this.application.applicationName,
             deploymentGroupName: props && props.deploymentGroupName,
             serviceRoleArn: this.role.roleArn,
             deploymentConfigName: props && props.deploymentConfig &&
                 props.deploymentConfig.deploymentConfigName,
+            autoScalingGroups: props && props.autoScalingGroups &&
+                props.autoScalingGroups.map(asg => asg.autoScalingGroupName()),
         });
 
         this.deploymentGroupName = resource.ref;
         this.deploymentGroupArn = deploymentGroupName2Arn(this.application.applicationName,
             this.deploymentGroupName);
     }
+
+    public get autoScalingGroups(): autoscaling.AutoScalingGroup[] {
+        return this._autoScalingGroups.slice();
+    }
+
+    private addCodeDeployAgentInstallUserData(agentInstallationType?: AgentInstallationType): void {
+        if (!agentInstallationType) {
+            return;
+        }
+
+        const stack = cdk.Stack.find(this);
+        const region = stack.requireRegion('Could not determine the region to download the CodeDeploy agent from');
+
+        for (const asg of this._autoScalingGroups) {
+            switch (agentInstallationType) {
+                case AgentInstallationType.UBUNTU_14_04:
+                    asg.addUserData(`
+                        apt-get update -y
```

Comment:
Not too keen on the embedded newlines here. Mind doing something like this:

```
asg.addUserData([
  'apt-get update -y',
  'apt-get install',
  ...
].join('\n'));
```

Reply:
Honestly we need to make `UserData` an object with a richer API.

Reply:
Or hide user data all together and opt for LaunchConfig :)

Reply:
> Honestly we need to make UserData an object with a richer API, and hide the platform differences in there as well.

I'll try to come up with something.